numerapi/numerapi.py

```diff
@@ -777,6 +784,80 @@ def upload_predictions(self, file_path: str, tournament: int = 8,
         submission_id = create['data']['create_submission']['id']
         return submission_id
 
+    def upload_predictions(self, df: pd.DataFrame, tournament: int = 8,
```

Comment:
overloading with a different signature doesn't really work in python. Have you tried your code? I am sure only one of the two options (path vs dataframe) works.